interfaces/HO-Portal/src/app/components/update-fsp/update-fsp.component.ts

```diff
@@ -0,0 +1,160 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { NgModel } from '@angular/forms';
+import { AlertController } from '@ionic/angular';
+import { TranslateService } from '@ngx-translate/core';
+import { ProgramsServiceApiService } from '../../services/programs-service-api.service';
+
+@Component({
+  selector: 'app-update-fsp',
+  templateUrl: './update-fsp.component.html',
+  styleUrls: ['./update-fsp.component.scss'],
+})
+export class UpdateFspComponent implements OnInit {
+  @Input()
+  public label: string;
+
+  @Input()
+  public explanation: string;
+
+  @Input()
+  public type: string;
+
+  @Input()
+  public value: string;
+
+  @Input()
+  public placeholder: string;
+
+  @Input()
+  public isDisabled: boolean;
+
+  @Input()
+  public inProgress: boolean;
+
+  @Input()
+  public fspList: any[];
+
+  @Input()
+  public referenceId: string;
+
+  @Output()
+  updated: EventEmitter<string> = new EventEmitter<string>();
+
+  public propertyModel: any | NgModel;
+
+  constructor(
+    private programsService: ProgramsServiceApiService,
+    private alertController: AlertController,
+    private translate: TranslateService,
+  ) {}
+
+  public startingAttributes: any[] = [];
+  public selectedFspAttributes: any[] = [];
+  public attributeDifference: any[] = [];
+  public startingFspName = '';
+  public selectedFspName = '';
+  public attributesToSave: {} = {};
+  public enableUpdateBtn = true;
+
+  ngOnInit() {
+    this.propertyModel = this.value;
+    this.startingFspName = this.value;
+    this.getFspAttributes(this.value);
+    this.startingAttributes = [...this.selectedFspAttributes];
+    this.enableUpdateBtn = false;
+  }
+
+  public updateChosenFsp() {
+    this.programsService
+      .updateChosenFsp(
+        this.referenceId,
+        this.selectedFspName,
+        this.attributesToSave,
+      )
+      .then(
+        () => {
+          this.inProgress = false;
+          this.actionResult(
+            this.translate.instant('common.update-success'),
+            true,
+          );
+        },
+        (error) => {
+          this.inProgress = false;
+          console.log('error: ', error);
+          if (error && error.error) {
+            this.actionResult(error.error.message);
```

Comment:
@elwinschmitz it was more difficult here to use the same formatErrors() function you've used elsewhere, because it's potentially about multiple attributes, etc. It could be done of course, but this was GEFN we judged.